include/dds/core/xtypes/detail/StructType.hpp

```diff
@@ -18,12 +18,147 @@
 #ifndef EPROSIMA_DDS_CORE_XTYPES_DETAIL_STRUCT_TYPE_HPP_
 #define EPROSIMA_DDS_CORE_XTYPES_DETAIL_STRUCT_TYPE_HPP_
 
+#include <string>
+#include <dds/core/xtypes/MemberType.hpp>
+
+#define COND_EXCEP_THROW(EXPR, CONT) if(EXPR) {\
+    throw IllegalOperationError(CONT) ;\
+}
+
+
 namespace dds {
 namespace core {
 namespace xtypes {
 namespace detail {
 
-class StructType {};
+
+class StructType
+{
+public:
+    StructType(){}
+
+    void name(
+            const std::string &s)
+    {
+        name_ = s ;
```

Comment:
eprosima style has no space before `;`
```suggestion
        name_ = s;
```